tests/deadcode.py

```diff
@@ -0,0 +1,269 @@
+            self.walk_annotation(node.args.posonlyargs, deadcode)
+            self.walk_annotation(node.args.args, deadcode)
+            self.walk_annotation(node.args.kwonlyargs, deadcode)
+
+            self.walk_annotation(node.args.vararg, deadcode)
+            self.walk_annotation(node.args.kwarg, deadcode)
+
+            self.check_childs(node.args.kw_defaults, deadcode)
+            self.check_childs(node.args.defaults, deadcode)
+
+            self.walk_annotation(node.returns, deadcode)
+            self.check_childs(node.decorator_list, deadcode)
+
+            self.check_stmts(node.body, deadcode)
+
+        elif isinstance(node, ast.ClassDef):
+            self.check_childs(node.decorator_list, deadcode)
+
+            self.check_childs(node.bases, deadcode)
+
+            self.check_childs(node.keywords, deadcode)
+
+            self.check_stmts(node.body, deadcode)
+
+        elif isinstance(node, ast.If):
+
+            test_value = self.static_value(node.test, deadcode)
+
+            if_is_dead = self.check_stmts(node.body, deadcode or (test_value == False))
+            else_is_dead = self.check_stmts(
+                node.orelse, deadcode or (test_value == True)
+            )
+            deadcode = if_is_dead and else_is_dead
+
+            self.walk_deadcode(node.test, (if_is_dead and not node.orelse) or deadcode)
+
+        elif isinstance(node, ast.Match):
+            self.walk_deadcode(node.subject, deadcode)
+            for _case in node.cases:
+                _case.deadcode = deadcode
+                self.walk_deadcode(_case.pattern, deadcode)
+                self.walk_deadcode(_case.guard, deadcode)
+
+            dead_cases = all(
+                [self.check_stmts(_case.body, deadcode) for _case in node.cases]
+            )
+
+            if any(
+                isinstance(_case.pattern, ast.MatchAs) and _case.pattern.pattern is None
+                for _case in node.cases
+            ):
+                # case _:
+                deadcode = dead_cases
+
+        elif isinstance(node, (ast.For, ast.AsyncFor)):
+            self.walk_deadcode(node.target, deadcode)
+            self.walk_deadcode(node.iter, deadcode)
+            self.check_stmts(node.body, deadcode)
+            self.check_stmts(node.orelse, deadcode)
+
+        elif isinstance(node, ast.IfExp):
+
+            test_value = self.static_value(node.test, deadcode)
+
+            if_is_dead = self.walk_deadcode(
+                node.body, deadcode or (test_value == False)
+            )
+            else_is_dead = self.walk_deadcode(
+                node.orelse, deadcode or (test_value == True)
+            )
+
+        elif isinstance(node, (ast.While)):
+            cnd = self.static_value(node.test, deadcode)
+
+            self.check_stmts(node.body, deadcode or cnd == False)
+            self.check_stmts(node.orelse, deadcode or cnd == False or cnd == True)
+
+            def contains_break(node):
```

Comment:
Please move this outside this function. It doesn't need to be local and the `node` variable is being shadowed.